prestodb/client.py

```diff
@@ -271,10 +272,13 @@ def max_attempts(self, value):
         with_retry = exceptions.retry_with(
             self._handle_retry,
             exceptions=(
-                exceptions.Http503Error,
+                KerberosExchangeError,
                 PrestoRequest.http.ConnectionError,
                 PrestoRequest.http.Timeout,
             ),
+            conditions=(
+                lambda x: x is not None and hasattr(x, 'status_code') and x.status_code == 503,
```

Comment:
I'd suggest `lambda response: getattr(response, 'status_code', None) == 503` and to add comment to explain why it is worth retrying.